vue-frontend/src/views/dashboard/index.vue

```diff
@@ -0,0 +1,509 @@
+        //     `Error fetching accounts: ${error.message}.
+        //     Code: ${error.code}. Data: ${error.data}`
+        //     );
+        // });
+        
+    },
+    methods: {
+        async onSubmit() {
+            let _this = this
+            if (_this.$web3.utils.isAddress(_this.ruleForm.address)) {
+                const allowedToWithdraw = await _this.$faucetContract.methods.allowedToWithdraw(_this.ruleForm.address).call()
+                
+                if (allowedToWithdraw) {
+                    try {
+                        // send request for tokens
+                        const response = await _this.sendRequest({ account: _this.ruleForm.address })
+                        //console.log(response)
+
+                        // if success, update balance and display tx_hash
+                        if (response.message === 'success') {
+                            await _this.timeout(3000)
+                            _this.amount_tip = false
+                            _this.txhash = response.tx_hash
+                            _this.active = 3
+                            _this.ruleForm.usdcBalance = parseFloat(_this.ruleForm.usdcBalance) + 100
+                        }
+                    } catch (err) {
+                        console.log('allowedToWithdraw err:', err)
+                        _this.tipIndex = 2
+                        _this.tipVisible = true
+                    }
+                } else {
+                    _this.tipIndex = 1
+                    _this.tipVisible = true
+                }
+            }
+        },
+        async sendRequest(jsonObject) {
+            let _this = this
+            try {
+                _this.transformVisible = true
+                const response = await axios.post(process.env.BASE_API, jsonObject, {
+                    headers: { 'Content-Type': 'application/json' },
+                })
+                return response.data
+            } catch (err) {
+                // Handle Error Here
+                _this.transformVisible = false
+                console.error(err)
+            }
+        },
+        timeout (delay) {
+            return new Promise((res) => setTimeout(res, delay))
+        },
+        async ethChange() {
+            let _this = this
+            if (_this.$web3.utils.isAddress(_this.ruleForm.address)) {
+                const matic = await _this.$web3.eth.getBalance(_this.ruleForm.address)
+                // console.log('matic:', matic)
+                // console.log(_this.$web3.utils.fromWei(matic, 'ether'))
+                _this.ruleForm.maticBalance = _this.$web3.utils.fromWei(matic, 'ether')
+
+                const tokens = await _this.$tokenContract.methods.balanceOf(_this.ruleForm.address).call()
+                // console.log('usdc:', tokens)
+                // console.log(_this.$web3.utils.fromWei(tokens, 'ether'))
+                _this.ruleForm.usdcBalance = _this.$web3.utils.fromWei(tokens, 'ether')
```

Comment:
on mumbai testnet, our USDC token has 18 decimal, so this one works
but on real polygon testnet, USDC token only has 6 decimal, may need another way to deal with tokens have different decimal